### PR TITLE
Gallery Home Page Sidebar Description Updates - RSC-1141

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-1141

### CODE REVIEW

File: gallery/src/pages/Home.tsx [hunk 1]
```diff
@@ -24,35 +28,157 @@ const Home = () =>
...
+        is grouped by general purpose, as follows:
       </NxP>
-      <NxH4>React Components</NxH4>
+      <NxDescriptionList>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Alerts and Indicators</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This section includes documentation regarding components which are intended to convey specific types of
+            information to the user. For example, alert components for informing the user of events, and counters, tags,
+            and threat indicators for creating consistent ways of present particular aspects of a data entity.
```
Comment (EmilyBermudez88) [inline gallery/src/pages/Home.tsx:40]:
```suggestion
            and threat indicators for creating consistent ways of presenting particular aspects of a data entity.
```

File: gallery/src/pages/Home.tsx [hunk 2]
```diff
@@ -24,35 +28,157 @@ const Home = () =>
...
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faCode} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Documentation for usage of bare CSS classes within HTML. Some RSC "components" consist almost entirely of
+            styling with little or no special behavior that must be implemented. For these cases, the bare classes
+            are themselves a documented and stable part of the RSC API, allowing them to be used even in applications
+            which do not use React. Many of these components are also exposed via React "convenience components"
+            which simply pair the class in question with the HTML tag to which it is most typically applied.
+            Convenience components are not typically documented separated and are simply mentioned on these non-React
```
Comment (EmilyBermudez88) [inline gallery/src/pages/Home.tsx:132]:
```suggestion
            Convenience components are not typically documented separately and are simply mentioned on these non-React
```

**Review by EmilyBermudez88 — APPROVED** [review]

File: gallery/src/pages/Home.tsx [hunk 3]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
         <NxH3>Navigating the Gallery</NxH3>
       </NxTile.SubsectionHeader>
       <NxP>
-        The sidebar at left can be used to navigate to the various examples, which are organized into groups
-        as follows:
+        The sidebar at left can be used to navigate to the various components, styles and utilities. The navigation
+        is grouped by general purpose, as follows:
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:32]: is grouped by the purpose of each element.

File: gallery/src/pages/Home.tsx [hunk 4]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
       </NxP>
-      <NxH4>React Components</NxH4>
+      <NxDescriptionList>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Alerts and Indicators</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This section includes documentation regarding components which are intended to convey specific types of
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:38]: This section includes documentation on components that convey specific types of

File: gallery/src/pages/Home.tsx [hunk 5]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            information to the user. For example, alert components for informing the user of events, and counters, tags,
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:39]: For e.g. alert components that indicate events, counters and tags.

File: gallery/src/pages/Home.tsx [hunk 6]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            and threat indicators for creating consistent ways of present particular aspects of a data entity.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: and threat indicatiors. <remove the rest of the sentence>

File: gallery/src/pages/Home.tsx [hunk 7]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Buttons and Dropdowns</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This section includes components that the user would activate in order to initiate some action – buttons,
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:46]: This section includes documentation on call to action components like buttons,

File: gallery/src/pages/Home.tsx [hunk 8]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            dropdowns and other menus, and also links.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:47]: dropdowns, menus and links.

File: gallery/src/pages/Home.tsx [hunk 9]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Forms</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This sections includes all components that are typically used within a form. Components for forms themselves,
+            form structures such as fieldset, and for various kinds of form fields all fall within this section.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: These include form structures like fieldset, and other form fields.

File: gallery/src/pages/Home.tsx [hunk 10]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Data Presentation</NxDescriptionList.Term>
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:58]: Data visualization

File: gallery/src/pages/Home.tsx [hunk 11]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          <NxDescriptionList.Description>
+            This section groups together components which are intended for organizing and displaying data. This includes
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:60]: This section includes documentation on how to use different components for data visualization in a consistent manner to represent numerical and hierarchical data.

File: gallery/src/pages/Home.tsx [hunk 12]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            views such as lists, tables, and trees, and also visualizations such as the donut chart.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:61]: These include lists, tables, trees and donut charts.

File: gallery/src/pages/Home.tsx [hunk 13]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Layout</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Components which make up the "page skeleton" are included in this section. This includes page-level components
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: This section includes documentation on how to use different page components that determine the look and feel and the general navigation, in a consistent manner.

File: gallery/src/pages/Home.tsx [hunk 14]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            such as the global sidebar and also containers such as tiles and accordions. In addition, rich examples of
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: These include global sidebars and containers such as tiles and accordions.

File: gallery/src/pages/Home.tsx [hunk 15]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            layouts incorporating multiple components working together are also found here.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: Remove this line.

File: gallery/src/pages/Home.tsx [hunk 16]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Typography and Icons</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This section includes semantic and typographic containers for text content, such as paragraphs and code
+            blocks.  It also includes utility components that facilitate the use of icons.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:76]: remove "blocks"

File: gallery/src/pages/Home.tsx [hunk 17]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>HTML Variants</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Some RSC components and styles have separate support for usage via React component and via bare CSS classes.
+            In these cases, the React components and bare CSS classes are documented separately. For any such component
+            which has both types of documentation, the HTML documentation is currently sequestered in this section.
+            Generally, for React-using consumers, the React components should be preferred over the bare CSS class
+            approach, and this section should thus generally be ignored.
+          </NxDescriptionList.Description>
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:87]: There are separate usage guidelines and documentation for some RSC components, depending on whether they are React components or bare CSS classes. For components that have both React and HTML/CSS documentation, React guidelines should be preferred and this section should be ignored.

File: gallery/src/pages/Home.tsx [hunk 18]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>HTML Variants</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Some RSC components and styles have separate support for usage via React component and via bare CSS classes.
+            In these cases, the React components and bare CSS classes are documented separately. For any such component
+            which has both types of documentation, the HTML documentation is currently sequestered in this section.
+            Generally, for React-using consumers, the React components should be preferred over the bare CSS class
+            approach, and this section should thus generally be ignored.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:86]: There are separate usage guidelines and documentation for some RSC components, depending on whether they are React components or bare CSS classes. For components that have both React and HTML/CSS documentation, React guidelines should be preferred and this section should be ignored.

File: gallery/src/pages/Home.tsx [hunk 19]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Mixins and Helpers</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            This section includes CSS and SCSS utilities such as helper classes, SCSS mixins, and CSS variables.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:92]: This section includes documentation on how to CSS and SCSS utilities like helper classes, SCSS mixins and CSS variables, in a consistent manner.

File: gallery/src/pages/Home.tsx [hunk 20]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+            components are present in this section, however React hooks are documented here in addition to JavaScript
+            functions and TypeScript types.
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>Guidelines</NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Rather than documentation of specific components or usable pieces of code, this section includes higher level
+            descriptions, tutorials, and recommendations regarding usage of RSC components and web UI development in
+            general.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: This section covers general documentation on web UI development, and tutorials, recommendations for using RSC components.

File: gallery/src/pages/Home.tsx [hunk 21]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
-        Explanations of broader layout concepts that includes groups of components and/or styles working together.
-      </NxP>
-      <NxH4>JavaScript &amp; TypeScript Utilities</NxH4>
-      <NxP>Utility function and datatypes available to users of RSC.</NxP>
+      <NxDescriptionList>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faReact} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Documentation for a React component or family of closely related React components. Typically includes
+            technical details about the props that may be passed to the component along with live examples of its use.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:121]: This section includes documentation for a React component or a family of related React components. It contains technical details with examples, on the properties that can be passed to the component.

File: gallery/src/pages/Home.tsx [hunk 22]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faCode} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Documentation for usage of bare CSS classes within HTML. Some RSC "components" consist almost entirely of
+            styling with little or no special behavior that must be implemented. For these cases, the bare classes
+            are themselves a documented and stable part of the RSC API, allowing them to be used even in applications
+            which do not use React. Many of these components are also exposed via React "convenience components"
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:130]: Suggesting reword for this sentence:  For these cases, the bare classes are themselves a documented and stable part of the RSC API, allowing them to be used even in applications which do not use React.

File: gallery/src/pages/Home.tsx [hunk 23]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faRulerCombined} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Complex, compound layout documentation. These pages feature combinations of RSC components that have
+            separately have their own documentation pages, but which must also be documented in unison in order to
+            understand the full effect. For instance, one such page demonstrates the use
+            of <NxCode>nx-read-only</NxCode>, <NxCode>nx-grid</NxCode>, and <NxCode>nx-tile</NxCode> together in order
+            to implement a common data display pattern.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:147]: Suggestion: This section contains documentation on the combination usage of RSC components, when used together. It explains how these components can be used together to display a common pattern.

File: gallery/src/pages/Home.tsx [hunk 24]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term>
+            <GalleryNavIconCSS3 />
+          </NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Documentation of CSS-based stylesheet utilities and helpers such as CSS custom properties (aka variables)
+            and general-purpose classes which may be applied to other RSC components or HTML elements in general.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx]: This section contains documentation of CSS-based stylesheet utilities and helpers such as CSS custom properties (aka variables) and general-purpose classes. The general purpose classes can be applied to other RSC components or HTML elements.

File: gallery/src/pages/Home.tsx [hunk 25]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faSass} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            Documentation of SCSS-based stylesheet utilities and helpers such as mixins. While projects are not
+            required to use SCSS in order to use RSC, any projects which do not use SCSS cannot directly use these
+            helpers.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:164]: This section contains documentation of SCSS-based stylesheet utilities and helpers such as mixins. These are required by projects that use SCSS based stylesheets.

File: gallery/src/pages/Home.tsx [hunk 26]
```diff
@@ -24,35 +27,157 @@ const Home = () =>
...
+          </NxDescriptionList.Description>
+        </NxDescriptionList.Item>
+        <NxDescriptionList.Item>
+          <NxDescriptionList.Term><NxFontAwesomeIcon icon={faJs} fixedWidth /></NxDescriptionList.Term>
+          <NxDescriptionList.Description>
+            JavaScript and TypeScript utilities. Documentation within these pages describes various functions,
+            data structures, and types that are available within RSC which can be valuable to a consuming application.
```
Comment (skatre) [inline gallery/src/pages/Home.tsx:171]: Suggestion: This section contains documentation on JavaScript and TypeScript Utlilites. It describes functions, data structures, and types that are available within RSC, to use by applications.

**Pull request merged**